Mathlib/Algebra/Polynomial/Splits.lean

```diff
@@ -322,6 +322,23 @@ theorem eq_prod_roots_of_splits_id {p : K[X]} (hsplit : Splits (RingHom.id K) p)
     p = C p.leadingCoeff * (p.roots.map fun a => X - C a).prod := by
   simpa using eq_prod_roots_of_splits hsplit
 
+theorem Splits.dvd_of_roots_le_roots {p q : K[X]} (hp : p.Splits (RingHom.id _)) (hp0 : p ≠ 0)
+    (hq : p.roots ≤ q.roots) : p ∣ q := by
+  by_cases hq0 : q = 0
+  · simp [hq0]
+  · rw [eq_prod_roots_of_splits_id hp, C_mul_dvd (leadingCoeff_ne_zero.2 hp0)]
+    exact dvd_trans
+      (Multiset.prod_dvd_prod_of_le (Multiset.map_le_map hq))
+      (prod_multiset_X_sub_C_dvd _)
+
+theorem Splits.dvd_iff_roots_le_roots {p q : K[X]}
+    (hp : p.Splits (RingHom.id _)) (hp0 : p ≠ 0) (hq0 : q ≠ 0) :
+    p ∣ q ↔ p.roots ≤ q.roots := by
+  refine ⟨?_, hp.dvd_of_roots_le_roots hp0⟩
+  rintro ⟨r, rfl⟩
+  rw [roots_mul hq0]
+  exact le_self_add
```

Comment:
```suggestion
  exact roots_mul hq0 ▸ le_self_add
```

Reply:
Or just:
```lean
theorem Splits.dvd_iff_roots_le_roots {p q : K[X]}
    (hp : p.Splits (RingHom.id _)) (hp0 : p ≠ 0) (hq0 : q ≠ 0) :
    p ∣ q ↔ p.roots ≤ q.roots :=
  ⟨fun ⟨_, hr⟩ ↦ hr ▸ roots_mul (hr ▸ hq0) ▸ le_self_add, hp.dvd_of_roots_le_roots hp0⟩
```

Reply:
I'm not convinced that we should encourage the triangle as a general policy

Reply:
Why not?

Reply:
Because it's a matter of the authors' style preferences. We have no goal to golf each proof. Some authors prefer to use the triangle, some authors prefer to split it into `rw` and `exact`, because it's more readable (at least, to them). Anyway, I think that this part of code will go away, see my suggestion above.